src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -141,6 +161,91 @@ const WorkflowIOTable = ({ which, inputsOutputs, config, errors, onChange, onSet
   ])
 }
 
+const BucketContentModal = ajaxCaller(class BucketContentModal extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      prefix: '',
+      objects: undefined
+    }
+  }
+
+  componentDidMount() {
+    this.load()
+  }
+
+  componentDidUpdate(prevProps) {
+    StateHistory.update(_.pick(['objects', 'prefix'], this.state))
+  }
+
+  async load(prefix = this.state.prefix) {
+    const { workspace: { workspace: { namespace, bucketName } }, ajax: { Buckets } } = this.props
+    try {
+      this.setState({ loading: true })
+      const { items, prefixes } = await Buckets.list(namespace, bucketName, prefix)
+      this.setState({ objects: items, prefixes, prefix })
+    } catch (error) {
+      reportError('Error loading bucket data', error)
+    } finally {
+      this.setState({ loading: false })
+    }
+  }
+
+  render() {
+    const { workspace: { workspace: { bucketName } }, onDismiss, onSelect } = this.props
+    const { prefix, prefixes, objects, loading } = this.state
+    const prefixParts = _.dropRight(1, prefix.split('/'))
+    return h(Modal, {
+      style: { flexGrow: 1, backgroundColor: 'white', border: `1px solid ${colors.gray[3]}`, padding: '1rem' },
+      activeStyle: { backgroundColor: colors.blue[3], cursor: 'copy' },
+      onDismiss,
+      title: 'Choose input file',
+      showX: true,
+      showButtons: false
+    }, [
+      div([
+        _.map(({ label, target }) => {
+          return h(Fragment, { key: target }, [
+            linkButton({ onClick: () => this.load(target) }, [label]),
+            ' / '
+          ])
+        }, [
+          { label: 'Files', target: '' },
+          ..._.map(n => {
+            return { label: prefixParts[n], target: _.map(s => `${s}/`, _.take(n + 1, prefixParts)).join('') }
+          }, _.range(0, prefixParts.length))
+        ])
+      ]),
+      div({ style: { margin: '1rem -1rem 1rem -1rem', borderBottom: `1px solid ${colors.gray[5]}` } }),
+      h(SimpleTable, {
```

Comment:
TOL: Is it worth using `SimpleTable` since there's only one column? All it's really giving us is the line divider styling.

Reply:
I personally really like the divider styling since the "directory" names can be pretty long and convoluted, having a visual way to separate them is nice.